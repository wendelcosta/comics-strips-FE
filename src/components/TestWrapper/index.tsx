import React from 'react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from '../../styles/theme'

const TestWrapper: React.FC = ({ children }: any) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export default TestWrapper
