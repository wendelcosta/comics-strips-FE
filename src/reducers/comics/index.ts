import * as ActionTypes from '../../actions/actionTypes'
import { ComicsState } from '../../actions/comicsTypes'

const initialState: ComicsState = {
  comics: [],
  isComicsLookUpPending: false,
}

const comicsReducer = (
  state: ComicsState = initialState,
  // eslint-disable-next-line
  action: any
): ComicsState => {
  switch (action.type) {
    case ActionTypes.GET_COMICS:
      return {
        ...state,
        comics: [],
        isComicsLookUpPending: true,
      }
    case ActionTypes.GET_COMICS_SUCCESS:
      return {
        ...state,
        comics: action.payload,
        isComicsLookUpPending: false,
      }

    default:
      return state
  }
}

export default comicsReducer
