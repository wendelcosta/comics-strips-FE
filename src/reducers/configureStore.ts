import { createStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { composeWithDevTools } from 'redux-devtools-extension'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

import configureRootReducer from '.'
import configureRootSaga from '../sagas'

const persistConfig = {
  key: 'Comics',
  storage,
}

const sagaMiddleware = createSagaMiddleware()
const middleware = [sagaMiddleware]

const persistedReducer = persistReducer(persistConfig, configureRootReducer())

const store = createStore(
  persistedReducer,
  composeWithDevTools(applyMiddleware(...middleware))
)
sagaMiddleware.run(configureRootSaga)
const persistor = persistStore(store)
export { persistor, store }
