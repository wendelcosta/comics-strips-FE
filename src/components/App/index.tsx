import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import CssBaseline from '@material-ui/core/CssBaseline'
import { ThemeProvider } from '@material-ui/core/styles'
import { Switch, Route } from 'react-router-dom'

import Container from '@material-ui/core/Container'

import Header from '../Header'
import Footer from '../Footer'
import Home from '../Pages/Home'
import Details from '../Pages/Details'
import ScrollToTop from '../ScrollToTop'

import theme from '../../styles/theme'

import { fetchComics } from '../../actions/comics'

const App: React.FC = () => {
  const dispatch = useDispatch()
  const [shouldFetch, setShouldFetch] = useState(true)
  useEffect(() => {
    // eslint-disable-next-line
    shouldFetch && dispatch(fetchComics())
    setShouldFetch(false)
  }, [dispatch, shouldFetch])

  const comics = useSelector(
    // @ts-ignore
    (state) => state.comics.comics
  )
  const isComicsLookUpPending = useSelector(
    // @ts-ignore
    (state) => state.comics.isComicsLookUpPending
  )

  return (
    <>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <ScrollToTop />
        <Header />
        <Container
          style={{ minHeight: '100%', marginBottom: 30 }}
          maxWidth="lg"
        >
          <Switch>
            <Route path="/details/:id">
              <Details />
            </Route>
            <Route path="/">
              <Home
                isComicsLookUpPending={isComicsLookUpPending}
                comics={comics}
                onReloadClick={(): any => (
                  // eslint-disable-next-line
                  setShouldFetch(true), dispatch(fetchComics())
                )}
              />
            </Route>
          </Switch>
        </Container>
        <Footer />
      </ThemeProvider>
    </>
  )
}

export default App
