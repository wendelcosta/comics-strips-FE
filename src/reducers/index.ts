import { combineReducers, Reducer } from 'redux'

import comics from './comics'

const configureRootReducer = (): Reducer =>
  combineReducers({
    comics,
  })
export default configureRootReducer
