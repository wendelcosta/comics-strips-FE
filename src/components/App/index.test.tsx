import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'

import TestWrapper from '../TestWrapper'
import { store } from '../../reducers/configureStore'

import App from '.'

describe('App Test', () => {
  test('Loads App', () => {
    const { container } = render(
      <Provider store={store}>
        <TestWrapper>
          <Router>
            <App />
          </Router>
        </TestWrapper>
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })
})
