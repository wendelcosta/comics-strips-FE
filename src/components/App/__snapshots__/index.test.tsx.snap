// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Loads App 1`] = `
<div>
  <header
    class="makeStyles-header-1"
  >
    <h1>
      Comics xkcd - Brought to you by Wendel Costa
    </h1>
  </header>
  <div
    class="MuiContainer-root-2 MuiContainer-maxWidthLg-8"
    style="min-height: 100%; margin-bottom: 30px;"
  >
    <div
      class="MuiGrid-root-13 makeStyles-gridContainer-10 MuiGrid-container-14 MuiGrid-spacing-xs-3-38"
    >
      <div
        class="makeStyles-loaderWrapper-11"
      >
        <div
          class="MuiCircularProgress-root-145 MuiCircularProgress-colorPrimary-148 MuiCircularProgress-indeterminate-147"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg-150"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle-151 MuiCircularProgress-circleIndeterminate-153"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <footer
    class="makeStyles-footer-144"
  >
    <p>
       
      2020
       Â© 
      Copyright Wendel Costa
    </p>
  </footer>
</div>
`;
